examples/passport-oauth2-login/src/authentication-strategies/strategy-adapter.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/authentication-passport
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {UserProfileFactory} from '@loopback/authentication';
+import {HttpErrors, Request} from '@loopback/rest';
+import {UserProfile} from '@loopback/security';
+import {Strategy} from 'passport';
+import {RedirectRoute} from './redirect-route';
+
+const passportRequestMixin = require('passport/lib/http/request');
+
+/**
+ * Adapter class to invoke passport-strategy
+ *   1. provides express dependencies to the passport strategies
+ *   2. provides shimming of requests for passport authentication
+ *   3. provides life-cycle similar to express to the passport-strategy
+ *   4. provides state methods to the strategy instance
+ * see: https://github.com/jaredhanson/passport
+ */
+export class StrategyAdapter<U> {
```

Comment:
Why is another strategy adapter being defined here? you should use: https://github.com/strongloop/loopback-next/blob/master/extensions/authentication-passport/src/strategy-adapter.ts .

Reply:
Were you waiting for https://github.com/strongloop/loopback-next/pull/4919 to land?

Reply:
hey @emonddr yes exactly, I was waiting for 4919 to be landed, the changes will have to be published as well.

Reply:
symlinks with master package is enabled now, please see PR description for manual test steps